src/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -817,6 +1009,39 @@ public void setPlotForSale(Resident resident, WorldCoord worldCoord, double forS
 			throw new TownyException(TownySettings.getLangString("msg_err_must_belong_town"));
 	}
 
+	public void setGroupForSale(Resident resident, PlotGroup group, double price) throws TownyException {
+		group.setPrice(price);
+
+		if (resident.hasTown()) {
+			try {
+
+				// exception
+				if (price > TownySettings.getMaxPlotPrice()) {
+					group.setPrice(TownySettings.getMaxPlotPrice());
+				} else {
+					group.setPrice(price);
+				}
+
+				if (price != -1) {
+					TownyMessaging.sendTownMessage(resident.getTown(), "Group " + group.getGroupName() + " was put for sale!");
+					if (group.getTown() != resident.getTown())
+						TownyMessaging.sendMessage(resident, "Group " + group.getGroupName() + " was put for sale!");
+				} else {
+					Player p = TownyAPI.getInstance().getPlayer(resident);
+					if (p == null) {
+						throw new TownyException("Player could not be found.");
+					}
+					p.sendMessage(TownySettings.getLangString("msg_plot_set_to_nfs"));
+					
+					// Since the groups are stored in towns we need to save the town.
+					TownyUniverse.getInstance().getDataSource().saveTown(group.getTown());
+				}
+			} catch (NotRegisteredException e) {
+					throw new TownyException(TownySettings.getLangString("msg_err_not_part_town"));
+			}
+		}
+	}
+

```

Comment:
Groups won’t be set for sale directly instead all of its plot will be set for sale